### ISSUE TITLE
v:null cannot travel for python binding

### ISSUE THREAD

**Issue by roxma** [vim/vim#2246]
```vim
:python3 vim.vars['foo'] = vim.vvars['null']
```

The above command will produce the following error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  TypeError: unable to convert NoneType to vim structure
```

I'm expecting `foo` to be set to `v:null`.

**Comment by k-takata** [comment]
Reproduced by:

```vim
:python3 vim.vars['foo'] = None
```

I think that the following patch fixes the problem:

```diff
--- a/src/if_py_both.h
+++ b/src/if_py_both.h
@@ -6231,6 +6231,11 @@ ConvertFromPyObject(PyObject *obj, typva
 
 	Py_DECREF(num);
     }
+    else if (obj == Py_None)
+    {
+	tv->v_type = VAR_SPECIAL;
+	tv->vval.v_number = VVAL_NULL;
+    }
     else
     {
 	PyErr_FORMAT(PyExc_TypeError,
```

**Comment by k-takata** [comment]
Hmm, this makes the test 86 and 87 fail.
https://bitbucket.org/k_takata/vim-ktakata-mq/addon/pipelines/home#!/results/125

**Comment by brammool** [comment]
Ken Takata wrote:

> Hmm, this makes the test 86 and 87 fail.
> https://bitbucket.org/k_takata/vim-ktakata-mq/addon/pipelines/home#!/results/125

Looks like it's all tests for None not being convertable.
We can fix the test.

We also have v:none, possibly better than v:null.

-- 
I am always surprised in the Linux world how quickly solutions can be
obtained.  (Imagine sending an email to Bill Gates, asking why Windows
crashed, and how to fix it...  and then getting an answer that fixed the
problem... <0>_<0> !)		              -- Mark Langdon

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by bfredl** [comment]
`v:null` corresponds better to `None` in python than `v:none`, both will encode to `null` in JSON for instance.

**Comment by k-takata** [comment]
Fixed by [8.0.1280](https://github.com/vim/vim/commit/de323093e1be165a3eadd1ab3d02942358cd97bf).
